x/ecocredit/base/keeper/msg_add_allowed_bridge_chain_test.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	"encoding/json"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/regen-network/gocuke"
+	"github.com/stretchr/testify/require"
+
+	types "github.com/regen-network/regen-ledger/x/ecocredit/base/types/v1"
+)
+
+type addAllowedBridgeChainSuite struct {
+	*baseSuite
+	err error
+}
+
+func TestAddAllowedBridgeChain(t *testing.T) {
+	gocuke.NewRunner(t, &addAllowedBridgeChainSuite{}).Path("./features/msg_add_allowed_bridge_chain.feature").Run()
+}
+
+func (s *addAllowedBridgeChainSuite) Before(t gocuke.TestingT) {
+	s.baseSuite = setupBase(t)
+}
+
+func (s *addAllowedBridgeChainSuite) ExpectChainNameToExist(a string) {
+	chainName := a
+	found, err := s.stateStore.AllowedBridgeChainsTable().Has(s.ctx, chainName)
+	require.NoError(s.t, err)
+	require.True(s.t, found)
+}
+
+func (s *addAllowedBridgeChainSuite) AllowedChainName(a string) {
+	_, err := s.k.AddAllowedBridgeChain(s.ctx, &types.MsgAddAllowedBridgeChain{
+		Authority: s.authority.String(),
+		ChainName: a,
+	})
+	require.NoError(s.t, err)
+}
```

Comment:
Might be nice to group `Expect` step functions together (i.e. `Then` step functions), as well as `Given` and `When`.